### PR TITLE
Revert "Set opt-level to 3"

### PR DESCRIPTION
**Pull request by SimonSapin**
This reverts commit aad9840ad49c56830384e87bc8bd87fd0199dc44.

Level 3 (possibly indirectly, the underlying bug might be in XCode’s linker) causes unit tests to sefault when compiled with some versions of XCode: https://github.com/rust-lang/rust/issues/50867

It also appears to cause some segfaults on Windows: https://github.com/rust-lang/rust/pull/50329#issuecomment-386853473, and regressions in some rustc performance benchmarks: https://github.com/rust-lang/rust/pull/50329#issuecomment-388084894

### CODE REVIEW

**Comment by SimonSapin** [comment]
Unfortunately, adding a test so that #50867 doesn’t regress again is tricky since the Travis-CI config uses newer XCode than versions that are affected according to https://github.com/rust-lang/rust/issues/50867#issuecomment-390435849.

**Review by kennytm — CHANGES_REQUESTED** [review]

File: src/Cargo.toml [hunk 1]
```diff
@@ -40,6 +40,14 @@ members = [
   "tools/rls/test_data/workspace_symbol",
 ]
 
+# Curiously, compiletest will segfault if compiled with opt-level=3 on 64-bit
+# MSVC when running the compile-fail test suite when a should-fail test panics.
+# But hey if this is removed and it gets past the bots, sounds good to me.
```
[the remark this answers is not in the document]
Comment (SimonSapin) [reply src/Cargo.toml]: Done.

**Review by kennytm — APPROVED** [review]

**Comment by nnethercote** [comment]
This regressed perf significantly, unfortunately:
http://perf.rust-lang.org/compare.html?start=59c0f5913ddc2f66c1ff8ab612f7027e38c85a6d&end=524ad9b9e03656f3fdeb03ed82fe78db3916e566&stat=instructions:u

**Pull request merged**